release-23.1: roachtest: clean up node shutdown infra and deflake c2c/shutdown

msbutler opened the pull request:
Backport 1/1 commits from #109227.

/cc @cockroachdb/release

Release justification: test infra change

---


The c2c/shutdown tests can flake if a node shuts down during cutover because
cutover may complete before the node shutdown infra has a chance to shut the
job down. This patch speeds the node infra shutdown logic in the following
ways:
- Previously, the c2c driver would poll every 5 seconds for the cutover phase.
  This patch adds a channel that will signal immediatly when cutover has begun.
- Previously the node shutdown infra would wait for all ranges to be fully
  replicated and wait an additional second before beginning shutdown. This
patch elides these checks for c2c node shutdown tests, as they can be shut down
immediatly after the `jobStarter` function returns.

This patch cleans up the roachtest node shutdown helper used by a variety of
tests. Notably, the patch removes one unecessary goroutine.

Informs #109119
Informs #108760

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/109362)
<!-- Reviewable:end -->

msbutler committed:
roachtest: clean up node shutdown infra and deflake c2c/shutdown

The c2c/shutdown tests can flake if a node shuts down during cutover because
cutover may complete before the node shutdown infra has a chance to shut the
job down. This patch speeds the node infra shutdown logic in the following
ways:
- Previously, the c2c driver would poll every 5 seconds for the cutover phase.
  This patch adds a channel that will signal immediatly when cutover has begun.
- Previously the node shutdown infra would wait for all ranges to be fully
  replicated and wait an additional second before beginning shutdown. This
patch elides these checks for c2c node shutdown tests, as they can be shut down
immediatly after the `jobStarter` function returns.

This patch cleans up the roachtest node shutdown helper used by a variety of
tests. Notably, the patch removes one unecessary goroutine.

Informs #109119
Informs #108760

Release note: None

msbutler commented:
pls don't mind the essential ci flake

msbutler commented:
unrelated ci flake. rerunning.
